src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1106,6 +1101,8 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 				this._onProcessExit(error);
 			}
 		});
+		//TODO:find better place for this
+		this._onIconChanged.fire(this);
```

Comment:
Maybe store `this._shellLaunchConfig.icon` before createProcess is called then and fire only if it's changed, can remove the todo then